rust/src/wrapper/device/mod.rs

```diff
@@ -230,35 +263,159 @@ impl Device {
     /// Returns the host used by the device, if any
     pub fn host(&mut self) -> PyResult<Option<Host>> {
         Python::with_gil(|py| {
-            self.0
+            self.obj
                 .getattr(py, intern!(py, "host"))
                 .map(|obj| obj.into_option(Host::from))
         })
     }
 
     /// Start advertising the data set with [Device.set_advertisement].
     pub async fn start_advertising(&mut self, auto_restart: bool) -> PyResult<()> {
+        if self.advertising_status == AdvertisingStatus::AdvertisingExtended {
```

Comment:
What about if the device is already advertising in legacy mode? Would that also be an error, or would it get overridden?

Reply:
Currently, bumble ignores the command if its already advertising. Added a comment with rationale for the current implementation.

```
// Bumble allows (and currently ignores) calling `start_advertising` when already
// advertising. Because that behavior may change in the future, we continue to delegate the
// handling to bumble.
```